File: 168. Excel Sheet Column Title/solution.js
/**
 * @param {number} n
 * @return {string}
 */
var convertToTitle = function(n) {
    if (n === 0) return '';
    
    var m = (n - 1) % 26; // 0 - 25
    return convertToTitle((n - m - 1) / 26) + '' + String.fromCharCode(m + 65);
};

File: 168. Excel Sheet Column Title/solution.cpp
class Solution {
public:
    string convertToTitle(int n) {
    string s = "";
    char c;
    
    while (n > 0) {
        int tail = n % 26;
        if (tail == 0) {    // 26(Z), 52(AZ), ... 
            tail = 26;
            c = 'Z';
        } else 
            c = tail + 'A' - 1;
        s = c + s;
        n -= tail;
        n /= 26;
    }
    
    return s;
}
};